client/src/featureform/deploy.py

```diff
@@ -0,0 +1,205 @@
+import os
+import platform
+import warnings
+from collections import namedtuple
+
+import docker
+import requests
+
+
+warnings.simplefilter("ignore")
+
+DOCKER_CONFIG = namedtuple(
+    "Docker_Config", ["name", "image", "port", "detach_mode", "env"]
+)
+
+
+class Deployment:
```

Comment:
nit: I would make this an abstract class, we don't ever want to instatiate one of these directly